src/storage/src/table/cell_based_table.rs

```diff
@@ -185,22 +186,70 @@ impl<S: StateStore> CellBasedTable<S> {
 
     pub async fn batch_write_rows(
         &mut self,
-        rows: Vec<(Row, Option<Row>)>,
+        rows: Vec<(Row, Row, Op)>,
         epoch: u64,
     ) -> StorageResult<()> {
         let mut batch = self.keyspace.state_store().start_write_batch();
         let mut local = batch.prefixify(&self.keyspace);
-        for (pk, cell_values) in rows {
+        let mut update_delete_keys = HashSet::new();
+        for (pk, cell_values, op) in rows {
             let arrange_key_buf =
                 serialize_pk(&pk, self.pk_serializer.as_ref().unwrap()).map_err(err)?;
-            let bytes = self
-                .cell_based_row_serializer
-                .serialize(&arrange_key_buf, cell_values, &self.column_ids)
-                .map_err(err)?;
-            for (key, value) in bytes {
-                match value {
-                    Some(val) => local.put(key, StorageValue::new_default_put(val)),
-                    None => local.delete(key),
+            match op {
+                Op::Delete => {
+                    let bytes = self
+                        .cell_based_row_serializer
+                        .serialize(&arrange_key_buf, Some(cell_values), &self.column_ids)
```

Comment:
I think the signature can be changed to directly accept a `Row`, instead of `Option<Row>`.

Reply:
... also the return value should only contain `Row` instead of `Option<Row>`. This is really confusing.